clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -267,11 +633,23 @@ else if (value instanceof String)
         HIGH, MEDIUM, LOW
     }
 
+    public enum Width {
+        NONE, SHORT, MEDIUM, LONG
+    }
+
+    /**
+     * Recommender interface
+     */
+    public interface Recommender {
+        List<Object> validValues(String name, Map<String, String> props);
```

Comment:
We probably need to document this interface with Javadocs. There are some important semantics that probably aren't obvious, e.g. that you are permitted to return an empty list.